Sources/StreamChat/Workers/ChannelListUpdater.swift

```diff
@@ -25,10 +25,10 @@ class ChannelListUpdater: Worker {
                 switch result {
                 case let .success(channelListPayload):
                     self?.database.write { session in
-                        
-                        if trumpExistingChannels {
-                            try session.deleteChannels(query: channelListQuery)
-                        }
+                        // TODO: this is not working well, we need a better approach to remove channels that are not in-sync
+//                        if trumpExistingChannels {
+//                            try session.deleteChannels(query: channelListQuery)
+//                        }
```

Comment:
Doesn't the safer DTO's resolve this? Or the crash still happens?

Reply:
it does not crash but it removes channels from the database that should not be removed (in my tests that breaks pagination) @dmigach FYI

Reply:
We can uncomment this. The reason why this was now not behaving correctly it is because of this line: https://github.com/GetStream/stream-chat-swift/pull/1379/files#diff-a233c9d8c6d1ea11dc5c3dd6e22e1db3cc35e3667a3a088e23aa9a2671483d03R292. The `delete` event is triggered and we should still call `deleteRows` but because the DTO fails to be created, we don't call it.